src/EditorFeatures/Core/ChangeSignature/AbstractChangeSignatureCommandHandler.cs

```diff
@@ -131,14 +137,15 @@ private static void HandleResult(ChangeSignatureResult result, Solution oldSolut
                 // wait context. That means the command system won't attempt to show its own wait dialog 
                 // and also will take it into consideration when measuring command handling duration.
                 context.OperationContext.TakeOwnership();
-                finalSolution = previewService.PreviewChanges(
+                finalSolution = await previewService.PreviewChangesAsync(
                     string.Format(EditorFeaturesResources.Preview_Changes_0, EditorFeaturesResources.Change_Signature),
                     "vs.csharp.refactoring.preview",
                     EditorFeaturesResources.Change_Signature_colon,
                     result.Name,
                     result.Glyph.GetValueOrDefault(),
                     result.UpdatedSolution,
-                    oldSolution);
+                    oldSolution,
+                    cancellationToken).ConfigureAwait(false);
```

Comment:
This also needs a ConfigureAwait(true) and/or SwitchToMainThread.

Reply:
It will need a switch, thread isn't guaranteed here prior to calling preview.